src/table/Root.tsx

```diff
@@ -1,31 +1,36 @@
 import React from 'react';
+import ReactDom from 'react-dom/client';
 import { StyleSheetManager } from 'styled-components';
 import { ThemeProvider } from '@mui/material/styles';
 import rtlPluginSc from 'stylis-plugin-rtl-sc';
+
 import TableWrapper from './components/TableWrapper';
 import { TableContextProvider } from './context';
 import muiSetup from './mui-setup';
+import { RenderProps, RootProps } from '../types';
 
-export function render(reactRoot, props) {
+export function render(reactRoot: ReactDom.Root | HTMLElement, props: RenderProps) {
```

Comment:
So since we have to take the native implementation into account it could be HTMLElement as well, and then cast it to ReactDom.Root? Now the file overriding thing doesn't seem like a good idea anymore

Reply:
When the native implementation runs:
`render(rootElement, { layout, model, manageData, theme, selectionsAPI, changeSortOrder, app, rect });`
reactRoot is HTMLElement

when our code runs:
```
 render(reactRoot, {
            rootElement,
            layout,
            tableData,
            direction,
            pageInfo,
            setPageInfo,
            constraints,
            translator,
            selectionsAPI,
            theme,
            changeSortOrder,
            keyboard,
            rect,
            footerContainer,
            announce,
          });
```
reactRoot is ReactDom.Root

No, it is really not.

Reply:
Well I guess we can do it like this for now. Better to write seperate functions with different names (like `renderNative` since they take different arguments at this point. Maybe we can start with your conversion in this PR and then we need to discuss a smarter way to use root

Reply:
Well I guess we can do it like this for now. Better to write seperate functions with different names (like `renderNative` since they take different arguments at this point. Maybe we can start with your conversion in this PR and then we need to discuss a smarter way to use root

Reply:
I tried your way before, it does not work for me.

Reply:
Actually, I tried many other ways and I realized that nebula implementation only do the current way.

Reply:
Not sure what you mean, I don't see why it wouldn't be possible. We can run different render functions depending on what env.carbon is. But that is not for this PR

Reply:
then when running `yarn build`, it complains.